document-store/src/test/java/org/hypertrace/core/documentstore/mongo/MongoCollectionTest.java

```diff
@@ -392,13 +392,19 @@ void testBulkUpdateWithFilter() throws IOException {
               query,
               List.of(dateUpdate, quantityUpdate, propsUpdate),
               UpdateOptions.DEFAULT_UPDATE_OPTIONS);
-
+      final ArgumentCaptor<com.mongodb.client.model.UpdateOptions> updateOptionsArgumentCaptor =
+          ArgumentCaptor.forClass(com.mongodb.client.model.UpdateOptions.class);
+      com.mongodb.client.model.UpdateOptions mongoUpdateOption =
+          new com.mongodb.client.model.UpdateOptions();
+      mongoUpdateOption.upsert(false);
       assertTrue(result.hasNext());
       assertJsonEquals(
           readFileFromResource("atomic_read_and_update/response.json").orElseThrow(),
           result.next().toJson());
 
-      verify(collection, times(1)).updateMany(filter, setObject);
+      verify(collection, times(1))
+          .updateMany(eq(filter), eq(setObject), updateOptionsArgumentCaptor.capture());
+      assertEquals(updateOptionsArgumentCaptor.getValue().toString(), mongoUpdateOption.toString());
```

Comment:
Any reason for comparing the `toString()`? This may not make the tests strict enough.

Reply:
UpdateOption of mongo client do not have equals implemented, it only has to string. So direct comparision is failing, Hence comparing using toString() method

Reply:
In that case, it's better to validate the fields that we are interested in (e.g.: `upsert` in this case) using getters. `toString()` based comparison doesn't check the datatype, etc.